aiojobs/_scheduler.py

```diff
@@ -10,20 +10,22 @@
     from collections.abc import Sized, Iterable, Container
     bases = Sized, Iterable, Container
 else:  # pragma: no cover
-    bases = (Collection,)
+    bases = (Collection, )
 
 
 class Scheduler(*bases):
-    def __init__(self, *, close_timeout, limit,
+    def __init__(self, *, close_timeout, limit, pending_limit,
                  exception_handler, loop):
         self._loop = loop
         self._jobs = set()
         self._close_timeout = close_timeout
         self._limit = limit
+        self._pending_limit = pending_limit
         self._exception_handler = exception_handler
         self._failed_tasks = asyncio.Queue(loop=loop)
         self._failed_task = loop.create_task(self._wait_failed())
         self._pending = deque()
+        self._waiting = deque()
```

Comment:
I believe using `asyncio.Queue` for `_pending` and not adding `_waiting` is more robust and simple solution.

Reply:
using `asyncio.Queue`  is simply, but can't limit the size of pending jobs , when set `pending_limit` == 0

Reply:
`asyncio.Queue(maxsize=pending_limit)`, infinite size if `pending_limit=0`.
Do you see any problem with this solution?

Reply:
I mean `spawn` always waits until one active job closes when `pending_limit=0`. And  the `_pending` has infinite size when `pending_limit=None`